Seda/Seda/Controllers/PaymentViewController.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  PaymentViewController.swift
+//  Seda
+//
+//  Created by Billy Chen on 5/17/20.
+//  Copyright © 2020 Billy Chen. All rights reserved.
+//
+
+import UIKit
+import Stripe
+import CardScan
+
+class PaymentViewController: UIViewController {
+    @IBOutlet weak var cardTextField: STPPaymentCardTextField!
+    @IBOutlet weak var cardNumberLabel: UILabel!
+    @IBOutlet weak var cashAmount: UITextField!
+    
+    let backendUrl = "http://localhost:4242/"
+    var scanStats: ScanStats?
+    var number: String?
+    var expiration: String?
+    var name: String?
+    var cardImage: UIImage?
+    var cashToSend: String = "0.00"
+    var paymentIntentClientSecret: String?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        print("viewDidLoad")
+        startCheckout()
+        //self.cardNumberLabel.text = format(number: self.number ?? "")
+        // Do any additional setup after loading the view.
+    }
+    
+    @IBAction func payPressed(_ sender: UIButton) {
+        pay()
+    }
+    
+    func format(number: String) -> String {
+        if number.count == 16 {
+            return format16(number: number)
+        } else if number.count == 15 {
+            return format15(number: number)
+        } else {
+            return number
+        }
+    }
+    
+    func format15(number: String) -> String {
+        var displayNumber = ""
+        for (idx, char) in number.enumerated() {
+            if idx == 4 || idx == 10 {
+                displayNumber += " "
+            }
+            displayNumber += String(char)
+        }
+        return displayNumber
+    }
+    
+    func format16(number: String) -> String {
+        var displayNumber = ""
+        for (idx, char) in number.enumerated() {
+            if (idx % 4) == 0 && idx != 0 {
+                displayNumber += " "
+            }
+            displayNumber += String(char)
+        }
+        return displayNumber
+    }
+    
+    func startCheckout() {
+        // Request a PaymentIntent from your server and store its client secret
+        // Create a PaymentIntent by calling the sample server's /create-payment-intent endpoint.
```

Comment:
What is a PyamentIntent? And is "your server" the flask server?

Reply:
A PaymentIntent is an object that holds all the data for making the payment. We make the PaymentIntent on the server and send back a unique key to the client that confirms everything is ok, and the client completes the transaction. I believe the reason that it's made in our server vs the client is for security reasons. It makes sure the client isn't able to send a different amount, etc. 

I'll make a design doc to go into detail on this.

Also yea the flask app is the server